src/sql/workbench/services/queryEditor/browser/queryEditorService.ts

```diff
@@ -91,6 +91,8 @@ export class QueryEditorService implements IQueryEditorService {
 		const m = await fileInput.resolve();
 		if (sqlContent) {
 			m.textEditorModel.setValue(sqlContent);
+			// prevent backup and restore of invalid sql upon restart.
```

Comment:
nit: But I also wouldn't use the phrase 'invalid sql' here. There's nothing invalid about this - we just don't care about preserving the query from edit data.

Maybe something like

`prevent backup and restore of backing query upon restart`

That being said - has there been discussion around re-opening the edit data editor upon restart? That seems to me to also be a reasonable solution - instead of opening a SQL editor it would open the entire editor (preserving any changes made to the query too). If you make an input type for edit data this shouldn't be too bad to do.